File: csharp/Ice/context/ContextI.cs
// Copyright (c) ZeroC, Inc.

using Demo;

internal class ContextI : ContextDisp_
{
    public override void call(Ice.Current current)
    {
        Console.Out.Write("Type = ");
        if (current.ctx.ContainsKey("type"))
        {
            Console.Out.Write(current.ctx["type"]);
        }
        else
        {
            Console.Out.Write("None");
        }
        Console.Out.WriteLine("");
    }

    public override void shutdown(Ice.Current current)
    {
        Console.Out.WriteLine("Shutting down...");
        current.adapter.getCommunicator().shutdown();
    }
}

File: java-compat/Ice/context/ContextI.java
// **********************************************************************
//
// Copyright (c) 2003-2016 ZeroC, Inc. All rights reserved.
//
// **********************************************************************

import Demo.*;

public class ContextI extends _ContextDisp
{
    @Override
    public void
    call(Ice.Current current)
    {
        System.out.print("Type = ");
        String type = current.ctx.get("type");
        if(type != null)
        {
            System.out.print(type);
        }
        else
        {
            System.out.print("None");
        }
        System.out.println();
    }

    @Override
    public void
    shutdown(Ice.Current current)
    {
        System.out.println("Shutting down...");
        current.adapter.getCommunicator().shutdown();
    }
}
